src/Psalm/Internal/Analyzer/Statements/Expression/BinaryOp/NonDivArithmeticOpAnalyzer.php

```diff
@@ -702,17 +821,24 @@ private static function analyzeNonDivOperands(
                     }
 
                     if ($parent instanceof PhpParser\Node\Expr\BinaryOp\Mod) {
-                        if ($always_positive) {
-                            if ($right_type_part instanceof TLiteralInt && $right_type_part->value === 1) {
-                                $result_type = Type::getInt(true, 0);
+                        if ($right_type_part instanceof TLiteralInt) {
+                            $literal_value_max = $right_type_part->value - 1;
```

Comment:
```suggestion
                            $literal_value_max = abs($right_type_part->value) - 1;
```


```
php > var_export(-5 % -3);
-2
```

Reply:
Thanks for the review! However, it would be easier to do it on this PR #6241. This one is based on #6241 so most of the changes are coming from there (including this one :) )